Simulation_engine/reforms.py

```diff
@@ -2,14 +2,237 @@
+    for nb_cp in [
+        "zeroChargePrincipale",
+        "unChargePrincipale",
+        "deuxOuPlusChargePrincipale",
+    ]:
+        if nb_cp not in parts_supp_cp:
+            raise LexCeption(
+                "the field '{}' is missing from 'partsParPACChargePartagee'. You can refer to the README to check valid format.".format(
+                    nb_cp
+                )
+            )
+
+    bonus_isole = calcul_nb_parts["bonusParentIsole"]
+
+    class nbptr(Variable):
+        value_type = float
+        entity = FoyerFiscal
+        label = "Nombre de parts"
+        reference = "http://vosdroits.service-public.fr/particuliers/F2705.xhtml"
+        definition_period = YEAR
+
+        def formula(foyer_fiscal, period, parameters):
+            """
+            Nombre de parts du foyer fiscal
+
+            note 1 enfants et résidence alternée (formulaire 2041 GV page 10)
+
+            quotient_familial.conj : nb part associées au conjoint d'un couple marié ou pacsé
+            quotient_familial.enf1 : nb part 2 premiers enfants
+            quotient_familial.enf2 : nb part enfants de rang 3 ou plus
+            quotient_familial.inv1 : nb part supp enfants invalides (I, G)
+            quotient_familial.inv2 : nb part supp adultes invalides (R)
+            quotient_familial.not31 : nb part supp note 3 : cases W ou G pour veuf, celib ou div
+            quotient_familial.not32 : nb part supp note 3 : personne seule ayant élevé des enfants
+            quotient_familial.not41 : nb part supp adultes invalides (vous et/ou conjoint) note 4
+            quotient_familial.not42 : nb part supp adultes anciens combattants (vous et/ou conjoint) note 4
+            quotient_familial.not6 : nb part supp note 6
+            quotient_familial.isol : demi-part parent isolé (T)
+            quotient_familial.edcd : enfant issu du mariage avec conjoint décédé;
+            """
+            nb_pac = foyer_fiscal("nb_pac", period)
+            maries_ou_pacses = foyer_fiscal("maries_ou_pacses", period)
+            celibataire_ou_divorce = foyer_fiscal("celibataire_ou_divorce", period)
+            veuf = foyer_fiscal("veuf", period)
+            jeune_veuf = foyer_fiscal("jeune_veuf", period)
+            # nbF = foyer_fiscal("nbF", period)
+            nbG = foyer_fiscal("nbG", period)
+            nbH = foyer_fiscal("nbH", period)
+            nbI = foyer_fiscal("nbI", period)
+            nbR = foyer_fiscal("nbR", period)
+            # nbJ = foyer_fiscal("nbJ", period)
+            # nbN = foyer_fiscal("nbN", period)  # noqa F841
+            caseP = foyer_fiscal("caseP", period)
+            caseW = foyer_fiscal("caseW", period)
+            caseG = foyer_fiscal("caseG", period)
+            caseE = foyer_fiscal("caseE", period)
+            caseK = foyer_fiscal("caseK", period)
+            caseN = foyer_fiscal("caseN", period)
+            caseF = foyer_fiscal("caseF", period)
```

Comment:
Traitons-nous vraiment le [caseF](https://fr.openfisca.org/legislation/caseF) ? Je ne vois pas le champ concerné dans la maquette.

Reply:
same, probablement supprimable j'ai pas fait le ménage dans la formule openfisca (certaines variables ne sont jamais utilisées même dans OF.